### PR TITLE
Catch symlink failure on Windows

### LINKED ISSUE #6480: Build failed on Windows
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**
Windows is not supported when running `Setup.py` 

```
19:37:50  powershell.exe : Traceback (most recent call last):
19:37:50  所在位置 C:\Workspace\Jenkins\workspace\mmdetection\build-on-windows\mmdetection@tmp\durable-961cdfce\powershellWrapper.ps1:3 字符: 1
19:37:50  + & powershell -NoProfile -NonInteractive -ExecutionPolicy Bypass -Comm ...
19:37:50  + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
19:37:50      + CategoryInfo          : NotSpecified: (Traceback (most recent call last)::String) [], RemoteException
19:37:50      + FullyQualifiedErrorId : NativeCommandError
19:37:50   
19:37:50    File ".\setup.py", line 182, in <module>
19:37:50      add_mim_extension()
19:37:50    File ".\setup.py", line 169, in add_mim_extension
19:37:50      os.symlink(src_relpath, tar_path)
19:37:50  OSError: [WinError 1314] �ͻ���û���������Ȩ��: '..\\..\\tools' -> '.\\mmdet\\.mim\\tools'
```

**Reproduction**

1. What command or script did you run?

```none
python setup.py develop
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?
3. What dataset did you use?

**Environment**
python3.8

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.
2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
If applicable, paste the error trackback here.

```none
A placeholder for trackback.
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

### PR DESCRIPTION
**Pull request by del-zhenwu**
Signed-off-by: del-zhenwu <dele.zhenwu@gmail.com>

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Fix #6480
## Modification
Try to catch the exception raised by `os.symlink` on Windows

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -166,7 +166,11 @@ def add_mim_extension():
 
             if mode == 'symlink':
                 src_relpath = osp.relpath(src_path, osp.dirname(tar_path))
-                os.symlink(src_relpath, tar_path)
+                try:
+                    os.symlink(src_relpath, tar_path)
+                except OSError:
+                    # Symlink fails with WinError on Windows
+                    shutil.copytree(src_relpath, tar_path)
```
Comment (RangiLyu) [inline setup.py:173]: It would be better to judge the `mode` by  `platform.system() == 'Windows'` in L142 instead of adding `try` here.

**Comment by RangiLyu** [comment]
Thanks for your contribution! Please sign the CLA first.
image

File: setup.py (hunk 1 above)
Comment (del-zhenwu) [reply setup.py]: solved

**Comment by RangiLyu** [comment]
Hi, the lint failed. Please install pre-commit and then run `pre-commit run --all-files` to fix.

**Comment by del-zhenwu** [comment]
The result of unit test seems to be is unstable.

**Comment by RangiLyu** [comment]
> The result of unit test seems to be is unstable.

It seems the reason of the latest version of MMCV. Sorry for that. We will solve that problem asap.

File: setup.py [hunk 2]
```diff
@@ -142,9 +143,11 @@ def add_mim_extension():
     if 'develop' in sys.argv:
```
Comment (hhaAndroid) [inline setup.py:143]: Logic should be:

```
if 'develop' in sys.argv :
  if platform.system() == 'Windows': 
      mode='copy'
  else:
     mode='sylink'

```

Comment (del-zhenwu) [reply setup.py:143]: I will make a fix

**Review by RangiLyu — APPROVED** [review]

**Comment by RangiLyu** [comment]
CI has been fixed in 2.18.1. Please pull master and rerun the tests.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**